addons/project/controllers/portal.py

```diff
@@ -224,6 +224,23 @@ def portal_my_project_task(self, project_id=None, task_id=None, access_token=Non
         task_sudo.attachment_ids.generate_access_token()
         values = self._task_get_page_view_values(task_sudo, access_token, project=project_sudo, **kw)
         values['project'] = project_sudo
+
+        task_url_common_start = "/my/project/%s/task/" % project_id
+        task_url_common_end = "?model=project.project&res_id={user_id}&access_token={access_token}".format(
+            user_id=values['user'].id, access_token=access_token)
+        history = request.session.get('my_project_tasks_history', [])
+
+        if task_id not in history:
+            return request.render("project.portal_my_task", values)
+
+        idx = history.index(task_id)
+        prev_task_id = idx != 0 and history[idx - 1]
+        next_task_id = idx < len(history) - 1 and history[idx + 1]
+        if prev_task_id:
+            values['prev_record'] = task_url_common_start + str(prev_task_id) + task_url_common_end
+        if next_task_id:
+            values['next_record'] = task_url_common_start + str(next_task_id) + task_url_common_end
```

Comment:
Don't we need to loop around when we are at the first or last task?
```py
prev_task_id = history[(idx - 1) % len(history)]
next_task_id = history[(idx + 1) % len(history)]
```
If yes, then `prev_record` and `next_record` always have a value, and we can do something like
```suggestion
        task_url = f"/my/project/{project_id}/task/%s?model=project.project&res_id={values['user'].id}&access_token={access_token}"
        values['prev_record'] = task_url % history[(idx - 1) % len(history)]
        values['next_record'] = task_url % history[(idx + 1) % len(history)]
```

Reply:
Thanks for the review, i made the change you suggested. I just modified it a bit for project that have only one task to remove the button, it does not make sense to me to leave them in that case.